client.go

```diff
@@ -614,11 +547,36 @@ func (c *TunaSessionClient) DialWithConfig(remoteAddr string, config *nkn.DialCo
 		if conn, ok := conns[connID(i)]; ok {
 			go func(conn *Conn, i int) {
 				defer conn.Close()
-				c.handleConn(conn, sessKey, i)
+				for {
+					c.handleConn(conn, sessKey, i)
+					if c.IsClosed() || sess.IsClosed() {
+						break
+					}
+
+					newPubAddrs, err := c.requestPubAddrs(remoteAddr, config)
+					if err != nil {
+						log.Printf("Request public addresses err %v", err)
+						newPubAddrs = pubAddrs // use previous address
+					}
+					if len(newPubAddrs.Addrs) >= i+1 {
+						conn, err = c.dialAConn(remoteAddr, sessionID, i, newPubAddrs.Addrs[i], config)
+						if err == nil {
+							continue
+						}
+						log.Printf("Tuna session conn %v re-dialAConn err: %v\n", i, err)
+					}
+					time.Sleep(2 * time.Second)
```

Comment:
We shouldn't handle conn if reconnect fails

Reply:
Yes, modified.